ui/pages/onboarding-flow/privacy-settings/privacy-settings.js

```diff
@@ -0,0 +1,130 @@
+import React, { useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import Box from '../../../components/ui/box';
+import Button from '../../../components/ui/button';
+import Typography from '../../../components/ui/typography';
+import ToggleButton from '../../../components/ui/toggle-button';
+import {
+  FONT_WEIGHT,
+  JUSTIFY_CONTENT,
+  TYPOGRAPHY,
+} from '../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import {
+  setFeatureFlag,
+  setUsePhishDetect,
+  setUseTokenDetection,
+} from '../../../store/actions';
+import { ONBOARDING_PIN_EXTENSION_ROUTE } from '../../../helpers/constants/routes';
+
+export default function PrivacySettings() {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const [usePhishingDetection, setUsePhishingDetection] = useState(true);
+  const [turnOnTokenDetection, setTurnOnTokenDetection] = useState(true);
+  const [showIncomingTransactions, setShowIncomingTransactions] = useState(
+    true,
+  );
+
+  const handleSubmit = () => {
+    dispatch(
+      setFeatureFlag('showIncomingTransactions', showIncomingTransactions),
+    );
+    dispatch(setUsePhishDetect(usePhishingDetection));
+    dispatch(setUseTokenDetection(turnOnTokenDetection));
+    history.push(ONBOARDING_PIN_EXTENSION_ROUTE);
+  };
+
```

Comment:
`renderSetting` can be extracted into small component of its own.

Reply:
Sure will do!

Reply:
Extracted @jpuri 😄